doc/nrf/security/crypto/drivers.rst

```diff
@@ -176,24 +178,31 @@ The following table provides an overview of the available directives and their c
    For the complete overview of the available configuration options, see the :ref:`ug_crypto_supported_features` page.
 
 The nRF Security subsystem checks the directives (set through the Kconfig options) to make the optimal driver selection.
-The subsystem combines the ``PSA_WANT_*`` and ``PSA_USE_*`` settings in Kconfig to make the appropriate driver selection for the cryptographic operation:
+The subsystem combines the settings in Kconfig that represent application requirements (``PSA_WANT_*``) and driver capabilities (``PSA_USE_*``) into ``PSA_NEED_*`` macros.
+These macros then control which drivers are compiled into the build and thus available for use at runtime.
+
+* If hardware acceleration is enabled and available for the requested features (as indicated by the relevant ``PSA_NEED_*`` macros), nRF Security selects that driver for performance and security reasons.
+* If no hardware driver is enabled and available, or if the hardware or the driver does not support the specific cryptographic operation (the corresponding ``PSA_NEED_*`` macro is not set for a hardware driver), nRF Security falls back to the software nrf_oberon driver, provided nrf_oberon's ``PSA_NEED_*`` macro is set.
 
-* If hardware acceleration is enabled and available for the requested features, nRF Security selects that driver for performance and security reasons.
-* If no hardware driver is enabled and available, or if the hardware or the driver does not support the specific cryptographic operation, nRF Security falls back to the software nrf_oberon driver.
+.. figure:: ../images/psa_want_use_need_flow.svg
+   :alt: Detailed process for feature and driver selection
+   :align: center
 
 For example:
 
 * You can configure the application to use CC310 and require the AES-192 key size through the :kconfig:option:`CONFIG_PSA_WANT_AES_KEY_SIZE_192` and :kconfig:option:`CONFIG_PSA_USE_CC3XX_KEY_MANAGEMENT_DRIVER` Kconfig options.
+  The resulting ``PSA_NEED_CC3XX_KEY_MANAGEMENT_DRIVER`` macro will be set if both are enabled.
```

Comment:
Thats not correct, if the user only wants an AES 192 bit key they PSA_NEED_CC3XX_KEY_MANAGEMENT_DRIVER will not be enabled.